test/metrics/metrics_test.go

```diff
@@ -0,0 +1,735 @@
+			})
+
+			userSignup1, err := hostAwait.UpdateUserSignupSpec(userSignupMember1.Name, func(us *toolchainv1alpha1.UserSignup) {
+				states.SetDeactivated(us, true)
+			})
+			require.NoError(t, err)
+			t.Logf("user signup '%s' set to deactivated", userSignupMember1.Name)
+			userSignup2, err := hostAwait.UpdateUserSignupSpec(userSignupMember2.Name, func(us *toolchainv1alpha1.UserSignup) {
+				states.SetDeactivated(us, true)
+			})
+			require.NoError(t, err)
+			t.Logf("user signup '%s' set to deactivated", userSignupMember2.Name)
+
+			t.Run("verify metrics", func(t *testing.T) {
+				metricsAssertion.WaitForMetricDelta(UserSignupsDeactivatedMetric, 2)                                 // two more because of deactivated users
+				metricsAssertion.WaitForMetricDelta(MasterUserRecordsPerDomainMetric, 0, "domain", "external")       // userSignupMember1 deactivated
+				metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", memberAwait.ClusterName)  // userSignupMember1 deactivated
+				metricsAssertion.WaitForMetricDelta(MasterUserRecordsPerDomainMetric, 0, "domain", "external")       // userSignupMember2 deactivated
+				metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", member2Await.ClusterName) // userSignupMember1 deactivated
+			})
+
+			t.Run("reactivate a deactivated user", func(t *testing.T) {
+				_, err := hostAwait.UpdateUserSignupSpec(userSignup1.Name, func(us *toolchainv1alpha1.UserSignup) {
+					states.SetDeactivating(us, false)
+					states.SetDeactivated(us, false)
+					states.SetApproved(us, true)
+				})
+				require.NoError(t, err)
+				_, err = hostAwait.UpdateUserSignupSpec(userSignup2.Name, func(us *toolchainv1alpha1.UserSignup) {
+					states.SetDeactivating(us, false)
+					states.SetDeactivated(us, false)
+					states.SetApproved(us, true)
+				})
+				require.NoError(t, err)
+
+				t.Run("verify metrics", func(t *testing.T) {
+					metricsAssertion.WaitForMetricDelta(UserSignupsMetric, 2)                                            // no change
+					metricsAssertion.WaitForMetricDelta(UserSignupsApprovedMetric, 4)                                    // two more because of reactivated user
+					metricsAssertion.WaitForMetricDelta(UserSignupsDeactivatedMetric, 2)                                 // no change
+					metricsAssertion.WaitForMetricDelta(MasterUserRecordsPerDomainMetric, 1, "domain", "external")       // userSignupMember1
+					metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 1, "cluster_name", memberAwait.ClusterName)  // userSignupMember1 is on member1
+					metricsAssertion.WaitForMetricDelta(MasterUserRecordsPerDomainMetric, 1, "domain", "external")       // userSignupMember2
+					metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 1, "cluster_name", member2Await.ClusterName) // userSignupMember2 is on member2
+				})
+			})
```

Comment:
this looks like a duplication of https://github.com/codeready-toolchain/toolchain-e2e/blob/900ccf701b2ab7e5bfc51f95fa91e2dac24562b7/test/metrics/metrics_test.go#L428-L461